lmnet/lmnet/networks/quantize_param_init.py

```diff
@@ -0,0 +1,101 @@
+import tensorflow as tf
+import functools
+
+
+class QuantizeParamInit:
+    """QuantizeParamInit.
+
+    This QuantizeParamInit class is a mixin class for quantized network of all kinds of task
+    (classification, object detection, and segmentation).
+    Every sub task's network class with quantized layer should extend this class.
+
+    Args:
+        activation_quantizer (callable): An activation quantization function
+        activation_quantizer_kwargs (dict): A dictionary of arguments for the activation quantization function
+        weight_quantizer (callable): A weight quantization function
+        weight_quantizer_kwargs (dict): A dictionary of arguments for the weight quantization function
+        quantize_first_convolution (boolean): True if using quantization at the first layer, False otherwise
+        quantize_last_convolution (boolean): True if using quantization at the last layer, False otherwise
+    """
+
+    def __init__(
+            self,
+            activation_quantizer=None,
+            activation_quantizer_kwargs=None,
+            weight_quantizer=None,
+            weight_quantizer_kwargs=None,
+            quantize_first_convolution=None,
+            quantize_last_convolution=None,
+            *args,
+            **kwargs
+    ):
+        super(QuantizeParamInit, self).__init__(*args, **kwargs)
+
+        assert callable(weight_quantizer)
+        assert callable(activation_quantizer)
+
+        activation_quantizer_kwargs = activation_quantizer_kwargs if activation_quantizer_kwargs is not None else {}
+        weight_quantizer_kwargs = weight_quantizer_kwargs if weight_quantizer_kwargs is not None else {}
+
+        weight_quantization = weight_quantizer(**weight_quantizer_kwargs)
+        self.activation = activation_quantizer(**activation_quantizer_kwargs)
+
+        self.custom_getter = functools.partial(self._quantized_variable_getter,
```

Comment:
This class depends on that `self.first_layer_name` and `self.last_layer_name` must be defined at aother place of network class, It looks much tight coupling.
Please implement unit test of the class for the suitable desing of this class.

Reply:
I implemented the unit test to confirm the definition of `self.first_layer_name` and `self.last_layer_name` as you suggested.

I also add the detail of class in the class document.

Reply:
Where is the unit test ?